usecases/scheduler/src/lambda/ssv2_configUpdater/configUpdater.py

```diff
@@ -0,0 +1,274 @@
+    existingSubscriptions = []
+    snsName = snsArnPrefix + '-' + ssv2configId
+
+    #create if not exists
+    resp = sns.create_topic(Name=snsName)
+    snsArn = resp.get('TopicArn')
+
+    args = {'TopicArn': snsArn}
+
+    while(True):
+        resp = sns.list_subscriptions_by_topic(**args)
+        existingSubscriptions.extend(resp.get('Subscriptions'))
+
+        NextToken = resp.get('NextToken')
+        if NextToken:
+            args['NextToken'] = NextToken
+        else:
+            break
+
+    for existingEmail in existingSubscriptions:
+        if existingEmail.get('Protocol') != 'email':
+            continue
+
+        if existingEmail.get('Endpoint') not in emails:
+            if existingEmail.get('SubscriptionArn') == 'PendingConfirmation':
+                print('Unable to remove {}, current status is PendingConfirmation. Amazon will delete automatically the subscription after 3 days of creation'.format(existingEmail.get('Endpoint')))
+                continue
+
+            print('Removing {}'.format(existingEmail.get('Endpoint')))
+            sns.unsubscribe(SubscriptionArn=existingEmail.get('SubscriptionArn'))
+
+        if existingEmail.get('Endpoint') in emails:
+            emails.remove(existingEmail.get('Endpoint'))
+
+    for email in emails:
+        print('Adding {}'.format(email))
+        sns.subscribe(
+            TopicArn=snsArn,
+            Protocol='email',
+            Endpoint=email
+        )
+    
+    pass    
+
+def deleteEventBridge(configId):
+    print("Attempting to delete EventBridge Scheduler: ...")
+
+    try:
+        scheduler.delete_schedule(
+        GroupName=schedulerGroupName,
+        Name='ScreenerScheduler-' + configId
+        )
+        return True
+    except botocore.exceptions.ClientError as e:
+        print(e.response['Error']['Message'])
+        return False
+
+def deleteSnsRecipient(ssv2configId):
+    # List all topics to find the one with the desired name
+    topics = []
+    next_token = None
+    topic_name = snsArnPrefix + '-' + ssv2configId
+    
+    while True:
+        if next_token:
+            response = sns.list_topics(NextToken=next_token)
+        else:
+            response = sns.list_topics()
+        
+        topics.extend(response['Topics'])
+        
+        if 'NextToken' in response:
+            next_token = response['NextToken']
+        else:
+            break
+    
+    # Find the topic with the specified name
+    topic_arn = None
+    for topic in topics:
+        if topic_name in topic['TopicArn']:
+            topic_arn = topic['TopicArn']
+            break
+    
+    if topic_arn is not None:
+        # Delete the topic by its ARN
+        sns.delete_topic(TopicArn=topic_arn)
+        print(f"Deleted topic: {topic_arn}")
+        return True
+    else:
+        print(f"Topic with name '{topic_name}' not found.")
+        return False
+
+
+
+
+def sanitizeEvent(event, action):
+    records = event['Records']
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;event&#39; (index: 0 | type: Unknown). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;action&#39;. The same parameter type is validated here for example: service-screener-v2/usecases/scheduler/src/lambda/ssv2_configUpdater/configUpdater.py:149. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness, emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles) using single or double underscore.

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).